R/system_file_linter.R

```diff
@@ -28,16 +28,19 @@ system_file_linter <- function() {
   funs <- c("system.file", "file.path")
   # either system.file(file.path(...)) or file.path(system.file(...))
   xpath_parts <- glue("
-  //SYMBOL_FUNCTION_CALL[text() = '{funs}']
-    /parent::expr[following-sibling::expr/expr/SYMBOL_FUNCTION_CALL[text() = '{rev(funs)}']]
+  parent::expr[following-sibling::expr/expr/SYMBOL_FUNCTION_CALL[text() = '{rev(funs)}']]
     /parent::expr
   ")
-  xpath <- paste(xpath_parts, collapse = " | ")
 
   Linter(linter_level = "expression", function(source_expression) {
-    xml <- source_expression$xml_parsed_content
-
-    bad_expr <- xml_find_all(xml, xpath)
+    xml_calls <- list(
+      source_expression$xml_find_function_calls(funs[1L]),
```

Comment:
readability has gone down a wee bit here, I don't think `funs` is likely to change in the future, with the new functionality I think it'll be clearer to just write out `"system.file"`/`"file.path"` more explicitly?

Reply:
okay, will do.